src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -1,221 +1,371 @@
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            Directory.CreateDirectory(Path.Combine(testDir.FullName, "AAA"));
+            Directory.CreateDirectory(Path.Combine(testDir.FullName, "AAAB"));
+            Directory.CreateDirectory(Path.Combine(testDir.FullName, "CAAA"));
+            using (File.Create(Path.Combine(testDir.FullName, "AAABB")))
+            using (File.Create(Path.Combine(testDir.FullName, "AAABBC")))
+            using (File.Create(Path.Combine(testDir.FullName, "CAAABB")))
+            {
+                if (TestFiles)
+                {
+                    string[] results = GetEntries(testDir.FullName, "AAABB");
+                    Assert.Equal(1, results.Length);
+                    Assert.Contains(Path.Combine(testDir.FullName, "AAABB"), results);
+                }
+                if (TestDirectories)
+                {
+                    string[] results = GetEntries(testDir.FullName, "AAA");
+                    Assert.Equal(1, results.Length);
+                    Assert.Contains(Path.Combine(testDir.FullName, "AAA"), results);
+                }
+            }
         }
-    }
 
-    [Fact]
-    public void SearchPatternNonFullPath()
-    {
-        //Shouldn't get files on full path by default
-        String testDir1Str = GetTestFileName();
-        DirectoryInfo testDir = new DirectoryInfo(TestDirectory);
-        testDir.CreateSubdirectory(testDir1Str);
-        using (File.Create(Path.Combine(TestDirectory, testDir1Str, GetTestFileName())))
+        [Fact]
+        public void SearchPatternIgnoreSubDirectories()
         {
-            Assert.Equal(1, GetEntries(TestDirectory, Path.Combine(testDir1Str, "*")).Length);
+            //Shouldn't get files on full path by default
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            Directory.CreateDirectory(Path.Combine(testDir.FullName, GetTestFileName()));
+            using (File.Create(Path.Combine(testDir.FullName, GetTestFileName())))
+            using (File.Create(Path.Combine(TestDirectory, GetTestFileName())))
+            {
+                string[] results = GetEntries(TestDirectory, Path.Combine(testDir.Name, "*"));
+                if (TestDirectories && TestFiles)
+                    Assert.Equal(2, results.Length);
+                else
+                    Assert.Equal(1, results.Length);
+            }
         }
-    }
 
-    [Fact]
-    public void SearchPatternIgnoreSubDirectories()
-    {
-        String testDir = GetTestFileName();
-        String testFile1 = Path.Combine(TestDirectory, GetTestFileName());
-        String testFile2 = Path.Combine(TestDirectory, testDir, GetTestFileName());
-        Directory.CreateDirectory(Path.Combine(TestDirectory, testDir));
-        using (File.Create(testFile1))
-        using (File.Create(testFile2))
+        [Fact]
+        [ActiveIssue(645)]
```

Comment:
Rather than this being an ActiveIssue, should this instead verify that an exception is thrown for a long path?  Then when long path support is added, the test should start failing and it can be fixed then?